packages/vaadin-combo-box/src/vaadin-combo-box-dropdown-wrapper.js

```diff
@@ -170,19 +135,58 @@ class ComboBoxDropdownWrapperElement extends PolymerElement {
 
       _itemIdPath: String,
 
-      /**
-       * Stores the scroller position before updating the 'items', in
-       * order to restore it immediately after 'items' have been updated
-       */
-      _oldScrollerPosition: {
-        type: Number,
-        value: 0
+      __effectiveItems: {
+        computed: '_getItems(opened, _items)',
+        observer: '__effectiveItemsChanged'
       }
     };
   }
 
   static get observers() {
-    return ['_loadingChanged(loading)', '_openedChanged(opened, _items, loading)', '_restoreScrollerPosition(_items)'];
+    return [
+      '_loadingChanged(loading)',
+      '_openedChanged(opened, _items, loading)',
+      '__updateAllItems(_selectedItem, renderer)'
+    ];
+  }
+
+  constructor() {
+    super();
+    this.__boundOnItemClick = this._onItemClick.bind(this);
+  }
+
+  __effectiveItemsChanged(effectiveItems) {
+    if (this.__virtualizer && effectiveItems) {
+      this.__virtualizer.size = effectiveItems.length;
+      this.__virtualizer.flush();
+    }
+  }
+
+  __createElements(count) {
+    return [...Array(count)].map(() => {
+      const item = document.createElement('vaadin-combo-box-item');
+      item.addEventListener('click', this.__boundOnItemClick);
+      item.tabIndex = '-1';
+      item.style.width = '100%';
+      return item;
+    });
+  }
+
+  __updateElement(el, index) {
+    const item = this.__effectiveItems[index];
+
+    el.setProperties({
+      item: this.__effectiveItems[index],
```

Comment:
We already get `this.__effectiveItems[index]` above, so can we use this:

```suggestion
      item: item,
```

Reply:
Done